### PR TITLE
[ci-skip][Docs]Add description for CTE to querying guide

### PR DESCRIPTION
**Pull request by hachi8833**
### Motivation / Background

This Pull Request has been created because the current querying guide has no entries for CTE and the link to `with` method.

### Detail

This Pull Request adds the entry for CTE.

### Additional information

ref: https://github.com/rails/rails/pull/37944
ref: https://github.com/rails/rails/pull/46843

cc: @vlado @palkan 

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`

### CODE REVIEW

File: guides/source/active_record_querying.md [hunk 1]
```diff
@@ -2162,6 +2162,57 @@ NOTE: Note that if a query matches multiple records, `find_by` will
 fetch only the first one and ignore the others (see the `LIMIT 1`
 statement above).
 
+### Using Common Table Expression (CTE)
+
+The [`with`][] method is a convenient way to access [Common Table Expressions (CTE)](https://en.wikipedia.org/wiki/Hierarchical_and_recursive_queries_in_SQL#Common_table_expression)'s WITH syntax, which can define named temporal query results that are durable and referable within a single SQL query.
```
Comment (zzak) [inline guides/source/active_record_querying.md:2167]:
```suggestion
The [`with`][] method is a convenient way to access [Common Table Expressions (CTE)](https://en.wikipedia.org/wiki/Hierarchical_and_recursive_queries_in_SQL#Common_table_expression) `WITH` syntax, which can define named temporal query results that are referable within a single SQL query.
```

"define named temporal query results " is still a hard statement to deconstruct, but I think this is reasonable

File: guides/source/active_record_querying.md [hunk 2]
```diff
@@ -2162,6 +2162,57 @@ NOTE: Note that if a query matches multiple records, `find_by` will
...
+
+NOTE: When using CTE with MySQL, the version of MySQL should be 8.0+.
```
Comment (zzak) [inline guides/source/active_record_querying.md:2169]:
```suggestion
NOTE: MySQL version 8.0+ is required to use CTEs
```

File: guides/source/active_record_querying.md [hunk 3]
```diff
@@ -2162,6 +2162,57 @@ NOTE: Note that if a query matches multiple records, `find_by` will
...
+
+The `with` method returns the instance of its `ActiveRecord::Relation`, so you can chain other query methods or refer to the defined query results from the chained methods including `from`, `select`, `joins` or `left_outer_joins`:
```
Comment (zzak) [inline guides/source/active_record_querying.md:2171]:
```suggestion
The `with` method returns an instance of `ActiveRecord::Relation`, so you can chain other query methods or refer to the defined query results from the chained methods including `from`, `select`, `joins` or `left_outer_joins`:
```

File: guides/source/active_record_querying.md [hunk 4]
```diff
@@ -2162,6 +2162,57 @@ NOTE: Note that if a query matches multiple records, `find_by` will
...
+SELECT posts.*, commented_posts.post_id AS has_comments
+FROM "posts"
+LEFT OUTER JOIN "commented_posts"
+  ON "commented_posts"."post_id" = "posts"."id"
+```
+
+You can also add multiple CTEs like the following:
+
+```ruby
+# Passing multiple key-value pairs:
```
Comment (zzak) [inline guides/source/active_record_querying.md:2196]:
```suggestion
You can also define multiple CTEs by passing multiple key-value pairs:

```ruby
```

File: guides/source/active_record_querying.md [hunk 5]
```diff
@@ -2162,6 +2162,57 @@ NOTE: Note that if a query matches multiple records, `find_by` will
...
+Post.with(
+  posts_with_comments: Post.where("comments_count > ?", 0),
+  posts_with_tags: Post.where("tags_count > ?", 0)
+)
+```
+
+```ruby
+# Or chaining multiple `with` calls
```
Comment (zzak) [inline guides/source/active_record_querying.md:2204]:
```suggestion
Or chaining multiple `with` calls:

```ruby
```

**Comment by hachi8833** [comment]
@zzak Applied all suggestions. Thank you very much! 😂

**Comment by zzak** [comment]
Can you squash please? 🙇

**Comment by hachi8833** [comment]
Looks like my forked rails repo and local repo has been something inconsistent, and the Rails repo might become mess if I'd squashed and pushed the branch...

To keep the Rails repo safe, I'd like to close the PR once and then create a new PR and branch with the reviewed document, referring to this PR. Shall I do that? @zzak

**Comment by zzak** [comment]
@hachi8833 I don't think there's a way for you to corrupt the Rails repo, anything you push to your branch will be rebased by GitHub, I think.

FWIW, I use this all the time to squash my branches and then select changes:
https://github.com/zzak/dotfiles/blob/45d385cafd0c3332423c9057546a86c515dd7d2d/zsh/.zshrc#L81-L82

I also have an `upstream` remote for https://github.com/rails/rails.

I'd be happy to help figure out the issue with you, it seems like this is not the first time it's happened. Feel free to ping me on ruby-jp slack if you have any questions :bow:

**Pull request closed without merging**